Add creature builder

Trollderim opened the pull request:
This pull request contains a new creature builder. It allows the simple creation of creatures with the suggested values from the GMG. The button to open the menu is at the top of npc sheets.

Current Features are:
- Selecting Perception, Saves, Abilities, Skills, ac, hp and strikes from suggested values
- Apply templates Brute, Skirmisher, Sniper and Soldier

The feature set atm is still limited. Future features should include:
- Add spell stuff (dc, attacks, spell casting stuff)
- Add more templates (spell casters, classes, skill paragon)

![CreatureBuilderAdvancedPrototyp](https://user-images.githubusercontent.com/7985081/110217063-62b15780-7eb2-11eb-9287-73b09b25e275.png)

Djphoenix719 commented:
Hey @Trollderim are you planning on adding the skills and strikes soon, or would you like to push this out now? I find a feature like this has great potential, but loses something if it's incomplete.

Trollderim commented:
> Hey @Trollderim are you planning on adding the skills and strikes soon, or would you like to push this out now? I find a feature like this has great potential, but loses something if it's incomplete.

I first wanted to push a version 0.1 for a first feedback and was planning on developing further this weekend. You can leave this PR open and I would get back in touch with you, if it is delayed.

Do you have any feedback on the existing stuff? I would do at least the skills similar. The strikes could get a little harder as I don't know how to add them from scratch without the data definition yet.

Trollderim commented:
I added some new features, see top post. 

@Djphoenix719 Do you think the missing features should be implemented before a release?

Djphoenix719 commented:
Cool everything looks good on my end. Skills is a good step for the first release. Let me know if all is ready and I can merge and push the build.

Djphoenix719 commented:
You can keep expanding as desired from here imo, enough there for v1, but up to you. Templates would be a great addition next maybe?

Trollderim commented:
I think I will take a quick shot tomorrow at the spells, maybe that's easier than expected. 

If it's harder I will clean some things up and will get in touch with you around 6pm GMT+1 I think.

Trollderim commented:
I added spells, but just really basic (DC and Attack). First I added the tradition and type too, but that made the code quite ugly. Those are not shown anyway in the sheet, so I removed them.

I think the v1 is complete now. I try to take a look into this project now and then for issues regarding the creature builder.

Features for a v2 could include:
- Class roadmaps
- Skill paragon roadmaps with different specializations
- A feature to add cool creature actions and/or spells to creatures by theme
